### PR TITLE
[fix][broker]Fixed produce and consume when anonymousUserRole enabled

### PR DESCRIPTION
**Pull request by tuteng**
My repo pr https://github.com/AmateurEvents/pulsar/pull/36

Fixes #xyz

Main Issue: #xyz

PIP: #xyz 

### Motivation

Implement the same logic as here https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L1010

When anonymousUserRole is enabled, pulsar-admin can successfully fetch cluster resources without authenticationProvider, but pulsar-client can't produce and consume data, this pr fixes that behavior, If the cluster is configured with anonymousUserRole, it should skip the authenticationProvider's checks

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [x] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Comment by tuteng** [comment]
Will update test and comment later

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java [hunk 1]
```diff
@@ -1050,9 +1050,13 @@ protected void handleConnect(CommandConnect connect) {
                         .getAuthenticationProvider(originalAuthMethod);
 
                 if (originalAuthenticationProvider == null) {
-                    throw new AuthenticationException(
-                            String.format("Can't find AuthenticationProvider for original role"
-                                    + " using auth method [%s] is not available", originalAuthMethod));
+                    authRole = getBrokerService().getAuthenticationService().getAnonymousUserRole()
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:1053]: Please add a comment to clarify the condition that has `connect.hasOriginalPrincipal()` but no `originalAuthMethod` means the proxy wants the broker to use anonymousUserRole. :)

**Review by mattisonchao — APPROVED** [review]
Please add a test for it. :)

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java (hunk 1 above)
Comment (tuteng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java:1059]: Fixed

**Comment by tuteng** [comment]
> Please add a test for it. :)

Fixed

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**